src/main/java/chess/controller/ChessController.java

```diff
@@ -69,30 +83,86 @@ private Command readCommand() {
+        outputView.printChessBoard(chessBoardDto);
+    }
+
+    private void endGame(ChessGame chessGame) {
+        renderChessBoard();
+        printStatus(chessGame);
+        if (chessGame.isSameStatus(GAME_OVER)) {
+            System.out.println("게임 종료 " + chessGame.getCurrentTeam() + "의 패배!");
+        }
+    }
+
     private void start(final List<String> commands) {
-        validateCommandSize(commands.size(), START);
-        if (!chessGame.isEnd()) {
+        final ChessGame chessGame = chessGameDao.select();
+        Command.validateCommandSize(commands.size(), START);
+        if (!isEnd(chessGame)) {
             throw new IllegalArgumentException("이미 체스 게임이 시작되었습니다.");
         }
+        chessGameDao.update(chessGame);
     }
 
     private void move(final List<String> commands) {
-        validateCommandSize(commands.size(), MOVE);
-        if (chessGame.isEnd()) {
+        final ChessGame chessGame = chessGameDao.select();
+        Command.validateCommandSize(commands.size(), MOVE);
+        if (isEnd(chessGame)) {
             throw new IllegalArgumentException("체스게임을 시작하려면 START를 입력하세요.");
         }
 
-        final String from = commands.get(MOVE_FROM_INDEX);
-        final String to = commands.get(MOVE_TO_INDEX);
+        PositionConvertor convertor = new PositionConvertor(commands);
+        chessGame.movePiece(convertor.getFromPosition(), convertor.getToPosition());
 
-        chessGame.movePiece(PositionConvertor.convert(from), PositionConvertor.convert(to));
+        chessGameDao.update(chessGame);
+    }
+
+    private void status(final List<String> commands) {
+        final ChessGame chessGame = chessGameDao.select();
+        Command.validateCommandSize(commands.size(), STATUS);
+        if (isEnd(chessGame)) {
+            throw new IllegalArgumentException("체스게임을 시작하려면 START를 입력하세요.");
+        }
+        printStatus(chessGame);
+    }
+
+    private void printStatus(final ChessGame chessGame) {
+        BigDecimal blackScore = chessGame.calculateScore(Team.BLACK);
+        outputView.printStatus(Team.BLACK, blackScore.longValue());
+
+        BigDecimal whiteScore = chessGame.calculateScore(Team.WHITE);
+        outputView.printStatus(Team.WHITE, whiteScore.longValue());
     }
 
     private void end(final List<String> commands) {
-        validateCommandSize(commands.size(), END);
-        if (chessGame.isEnd()) {
+        final ChessGame chessGame = chessGameDao.select();
+        Command.validateCommandSize(commands.size(), END);
+        if (isEnd(chessGame)) {
             throw new IllegalArgumentException("체스게임을 시작하려면 START를 입력하세요.");
         }
+        chessGameDao.update(chessGame);
+    }
+
+    private boolean isEnd(final ChessGame chessGame) {
+        GameStatus status = chessGame.getStatus();
+        return status == IDLE || status == GameStatus.GAME_OVER;
     }
 
     private <T> T readValidateInput(final Supplier<T> function) {
```

Comment:
```suggestion
        Optional<T> input = repeatByEx(function);
        while (input.isEmpty()) {
            input = repeatByEx(function);
        }
        return input;
```
do-while도 좋지만  초기화와 선언이 동시에 일어나도록 위와같이 수정하는 방법도 존재합니다(사실 큰 상관은 없습니다)